zetta_utils/db_annotations/annotation.py

```diff
@@ -0,0 +1,157 @@
+"""Manage annotations in a DB Layer."""
+
+from __future__ import annotations
+
+from typing import cast
+
+from neuroglancer.viewer_state import (
+    AxisAlignedBoundingBoxAnnotation,
+    EllipsoidAnnotation,
+    LineAnnotation,
+    PointAnnotation,
+)
+from typing_extensions import TypeAlias
+
+from zetta_utils.layer.db_layer import DBRowDataT, build_db_layer
+from zetta_utils.layer.db_layer.datastore import DatastoreBackend
+from zetta_utils.parsing.ngl_state import AnnotationKeys
+
+from . import constants
+
+NgAnnotation: TypeAlias = (
+    AxisAlignedBoundingBoxAnnotation | EllipsoidAnnotation | LineAnnotation | PointAnnotation
+)
+
+INDEXED_COLS = ("collection", "layer_group", "tags")
+NON_INDEXED_COLS = (
+    "comment",
+    AnnotationKeys.TYPE.value,
+    AnnotationKeys.POINT.value,
+    AnnotationKeys.POINT_A.value,
+    AnnotationKeys.POINT_B.value,
+    AnnotationKeys.CENTER.value,
+    AnnotationKeys.RADII.value,
+)
+
+DB_NAME = "annotations"
+DB_BACKEND = DatastoreBackend(DB_NAME, project=constants.PROJECT, database=constants.DATABASE)
+DB_BACKEND.exclude_from_indexes = NON_INDEXED_COLS
+ANNOTATIONS_DB = build_db_layer(DB_BACKEND)
+
+
+def read_annotation(annotation_id: str) -> DBRowDataT:
```

Comment:
Do we want to give all these functions an optional `annotation_db` argument that defaults to `ANNOTATIONS_DB` to make it testable?